### PR TITLE
Add a new TabletFile interface

### PR DESCRIPTION
**Pull request by cshannon**
This is a follow on to #3449 and adds a new `TabletFile` interface that can be used instead of having to use the abstract class `AbstractTabletFile` everywhere.

### CODE REVIEW

**Review by EdColeman — APPROVED** [review]

**Review by EdColeman — COMMENTED** [review]
Changes look okay - the is a check-style issue with TabletFile missing the header that fails the build.

File: core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java:23]: The type parameter T is not used, could it be removed?

```suggestion
public interface TabletFile {
```

Comment (cshannon) [reply core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java:23]: Good point, I just copied that from AbstractTabletFile where it was used for Comparable so I can drop it.

**Review by EdColeman — APPROVED** [review]
Assuming build passes - things look okay.

**Review by keith-turner — APPROVED** [review]

**Comment by cshannon** [comment]
I will merge this when QA checks pass. Also if someone has a better idea to rename the classes in the future that is fine too, I mostly just wanted to make TabletFile an interface since it's a generic name and then have more specific names for the other concrete TabletFile types which this PR and #3449 does. But the exact names aren't a big deal.

File: core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java:33]: Could you expand on these javadocs a bit to emphasize the difference between these two methods? It's not at all clear. Also, the `@return` is probably not really needed, if you just put a good main description.

**Review by ctubbsii — COMMENTED** [review]
Nice improvement. It also makes it clear that we don't care about sortability/comparability in these cases.

I'm commenting after this got merged, so my suggestion for better javadoc could be done as a follow-on.

File: core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java:33]: Created #3455 and assigned to you so it doesn't get lost. If you don't want to / can't do it, just unassign yourself, and somebody else can pick it up.

**Pull request merged**
